src/xcode/ENA/ENA/Source/Services/CoronaTestService/CoronaTest.swift

```diff
@@ -106,31 +106,64 @@ enum CoronaTest {
 		}
 	}
 
+	var pcrTest: PCRTest? {
+		switch self {
+		case .pcr(let test):
+			return test
+		case .antigen:
+			return nil
+		}
+	}
+	
+	var antigenTest: AntigenTest? {
+		switch self {
+		case .pcr:
+			return nil
+		case .antigen(let test):
+			return test
+		}
+	}
 }
 
-struct PCRTest: Codable, Equatable {
+protocol Test: Equatable {
+	
+	var registrationToken: String? { get }
+
+	var testResult: TestResult { get }
+	var testResultReceivedDate: Date? { get }
+	var positiveTestResultWasShown: Bool { get }
+
+	var isSubmissionConsentGiven: Bool { get }
+	// Can only be used once to submit, cached here in case submission fails
+	var submissionTAN: String? { get }
+	var keysSubmitted: Bool { get }
+
+	var journalEntryCreated: Bool { get }
+}
+
+struct PCRTest: Test, Codable {
```

Comment:
Is this protocol used somewhere?

Reply:
Yes, in CoronaTest, PCRTest and AntigenTest.

Reply:
Ah, yes, I was just wondering If somewhere else we rely on their conformance. I don't understand what the upside of having this protocol is. Is it just to keep the property naming in sync? Could we make use of it somewhere else?

Reply:
I added it because I saw the same properties being used in those three enum/structs. This way we could just rely on the protocol in the future and not have to differentiale between PCR and Antigen test.

Reply:
My intention with the CoronaTest enum with all the convenience properties was already that we do not have to differentiate between the two test types in a lot of cases, so if I understand it correctly, it kind of fulfills the same purpose as the protocol would. My suggestion would be to only introduce a protocol as another abstraction in case we need it from some other point in the app where it makes handling the different kind of tests easier for us.

Reply:
But isn’t this what protocols are for? I would suggest to leave the protocol and remove the enum to stay conform with the language philosophy.